struct UrlParams {
  1: required map<string, string> url_kvs,
  2: required string uri,
  3: required string query,
  4: required list<string> query_splits,
  5: required list<string> kv_pairs,
}
